docs/src/developers_guide/contributing_pytest_conversions.rst

```diff
@@ -41,7 +41,8 @@ Conversion Checklist
 
 #. Check for references to ``@tests``. These should be changed to ``@_shared_utils``.
 #. Check for references to ``with mock.patch("...")``. These should be replaced with
-   ``mocker.patch("...")``. ``mocker`` is a fixture, and can be passed into functions.
+   ``with mocker.patch("...")``. ``mocker`` is a fixture, should be passed into
+   relevant functions as a parameter.
```

Comment:
In many cases this is not necessary, since the `mocker` fixture includes appropriate teardown after `yield`. You would need to use the context manager syntax in two cases I can think of:

- If you need to reference the object (e.g. `x` in your example below). Often mocking is just modifying a module or similar so nothing needs to be referenced.
- The mocking needs to be temporary even within the test. Maybe you're mocking in two different ways and each one therefore needs to be temporary.

Perhaps the simplest solution is something like:


```suggestion
   ``mocker.patch("...")`` or ``with mocker.patch("...")``. ``mocker`` is a fixture, should be passed into
   relevant functions as a parameter.
```